File: backend/src/domain/entities/masters/question_option.py
"""QuestionOption Master — domain entity."""

from dataclasses import dataclass, field

from src.domain.entities.base_entity import BaseEntity


@dataclass
class QuestionOption(BaseEntity):
    """Option associated with a Question master."""

    option_title: str = field(default="")
    is_response_option: bool = field(default=False)
    question_id: int = field(default=0)
    is_active: bool = field(default=True)

File: frontend/src/features/masters/models/QuestionOption.ts
import type { MasterRecord } from './common';

export interface QuestionOption extends MasterRecord {
  option_title: string;
  is_response_option: boolean;
  question_id: number;
}

export interface CreateQuestionOptionRequest {
  option_title: string;
  is_response_option: boolean;
  question_id: number;
  is_active: boolean;
}

export type UpdateQuestionOptionRequest = Partial<CreateQuestionOptionRequest>;
